rotkehlchen/tests/db/test_db.py

```diff
@@ -1274,3 +1288,177 @@ def test_int_overflow_at_tuple_insertion(database, caplog):
     assert len(errors) == 1
     assert 'Failed to add "asset_movement" to the DB with overflow error' in errors[0]
     assert 'Overflow error while trying to add "asset_movement" tuples to the DB. Tuples:' in caplog.text  # noqa: E501
+
+
+def test_all_location_in_db(database):
+    """
+    Test that all locations in DB deserialize to a valid Location
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    locations = cursor.execute("SELECT location,seq from location")
+
+    # We deserialize, then serialize and compare the result
+    for location_letter, seq in locations:
+        deserialized_location = deserialize_location_from_db(location_letter)
+        assert deserialized_location.value == seq
+        assert Location(seq).serialize_for_db() == location_letter
+        location_name = deserialize_location(str(deserialized_location))
+        assert location_name == deserialized_location
+
+
+def test_all_locations_are_present(database):
+    """
+    Test that all Location are present in db
+    """
+    cursor = database.conn.cursor()
+    for location in Location:
+        r = cursor.execute("SELECT EXISTS(SELECT 1 FROM location WHERE seq=?)", (location.value,))
+        assert r.fetchone() == (1,)
+
+
+def test_all_trade_types_in_db(database):
+    """
+    Test that all trade_types in DB deserialize to a valid TradeType
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    trade_types = cursor.execute("SELECT type, seq from trade_type")
+
+    # We deserialize, then serialize and compare the result
+    for trade_type, seq in trade_types:
+        deserialized_trade_type = deserialize_trade_type_from_db(trade_type)
+        assert deserialized_trade_type.value == seq
+        assert TradeType(seq).serialize_for_db() == trade_type
+        trade_type_name = deserialize_trade_type(str(deserialized_trade_type))
+        assert trade_type_name == deserialized_trade_type
+
+
+def test_all_trade_types_are_present(database):
+    """
+    Test that all TradeType are present in db
+    """
+    cursor = database.conn.cursor()
+    for trade_type in TradeType:
+        r = cursor.execute(
+            "SELECT EXISTS(SELECT 1 FROM trade_type WHERE seq=?)",
+            (trade_type.value,))
+        assert r.fetchone() == (1,)
+
+
+def test_all_action_types_in_db(database):
+    """
+    Test that all action_type in DB deserialize to a valid ActionType
+    """
+    # Query for all locations
+    cursor = database.conn.cursor()
+    action_types = cursor.execute("SELECT type, seq from action_type")
+
+    # We deserialize, then serialize and compare the result
+    for action_type, seq in action_types:
+        deserialized_action_type = deserialize_action_type_from_db(action_type)
+        assert deserialized_action_type.value == seq - 1
```

Comment:
I see where this is coming from but probably it would not hurt to change all enums that have DB to start from `1`. For consistency's sake. Kind of sucks, we started the DB enum counting from 1 and not 0, but now it's done and DB does not change easily (would need an upgrade).

But the enums can change just in the code.